### DISCUSSION TITLE
I can't install Companion App on ubuntu 22.04 for firefox video download, can anyone help me please ????

### DISCUSSION THREAD

**Discussion by roma-ar** [aclap-dev/video-downloadhelper#420] [Help and Q&A]
### Your question

I tried to install it with command in terminal, with file.deb but without success.
Or problem is that the software video-download can't see Companion App

![download-video](https://github.com/aclap-dev/video-downloadhelper/assets/90523071/63319f42-4e5f-4620-8b29-96e2f5da5997)

After recheck nothing has changed
Restart browser, and sudo apt update without success

It's installed you can see in folder on photo:

![download-video_2](https://github.com/aclap-dev/video-downloadhelper/assets/90523071/f0ade2de-9a8d-47cd-8ce9-6f23267c105c)

About version Firefox:

![download-video_3](https://github.com/aclap-dev/video-downloadhelper/assets/90523071/00bd478e-0f66-4181-b04e-753f6834d280)

### Addon & CoApp version and details

I installed:  
downloadhelper.coapp-1.6.3-1_amd64.deb, 
and this : downloadhelper.coapp-1.6.3-1_amd64.tar.gz, 
and this : vdhcoapp-linux-x86_64.deb, 
and this : vdhcoapp-2.0.10-linux-x86_64.deb

**Comment by paulrouget** [comment]
That's a very old coapp. Where did you get these?

Run this command:

```
curl -sSLf https://github.com/aclap-dev/vdhcoapp/releases/latest/download/install.sh | bash
```

Then if it doesn't work, do this:

https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#ubuntu-and-flatpak--snap-based-systems

**Comment by EmilR73** [comment]
This is because Ubuntu uses Snap based Firefox!
I found a solution here:
https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized
Or you could install Firefox as a DEB:
https://www.omgubuntu.co.uk/2022/04/how-to-install-firefox-deb-apt-ubuntu-22-04

**Comment by roma-ar** [comment]
Thank you a lot 
This was helped me :https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#ubuntu-and-flatpak--snap-based-systems
